src/ray/core_worker/transport/concurrency_group_manager.h

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+namespace ray {
+
+namespace core {
+
+/// A manager that manages a set of concurrency group executors, which will perform
+/// the methods defined in one concurrency group.
+template <typename ConcurrencyGroupExecutorType>
+class ConcurrencyGroupManager final {
```

Comment:
Why do you implement everything in the header? Prefer implement this in .cc

Reply:
Goood catch, DONE.